File: node.js/src/original/domain/books/dtos/bookDetails.ts
export type BookDetails = {
  id: string;
  currentState: string;
  title: string;
  author: AuthorDetails;
  publisherName: string;
  edition: number;
  genre?: string | undefined;
  isbn?: string | undefined;
  publicationDate?: string | undefined;
  totalPages?: number | undefined;
  numberOfIllustrations?: number | undefined;
  bindingType?: string | undefined;
  summary?: string | undefined;
  committeeApproval: CommitteeApprovalDetails | undefined;
  reviewers: string[];
  chapters: ChapterDetails[];
  translations: TranslationDetails[];
  formats: FormatDetails[];
};

export type AuthorDetails = { firstName: string; lastName: string };
export type CommitteeApprovalDetails = {
  isApproved: boolean;
  feedback: string;
};
export type ChapterDetails = { title: string; content: string };
export type TranslationDetails = { language: string; translator: string };
export type FormatDetails = {
  formatType: string;
  totalCopies: number;
  soldCopies: number;
};

File: csharp/Slimmed/PublishingHouse.Persistence/Books/DTOs/BookDetails.cs
namespace PublishingHouse.Persistence.Books.DTOs;

using static BookDetails;

public record BookDetails(
    Guid Id,
    string CurrentState,
    string Title,
    AuthorDetails Author,
    string PublisherName,
    int Edition,
    string? Genre,
    string? ISBN,
    DateOnly? PublicationDate,
    int? TotalPages,
    int? NumberOfIllustrations,
    string? BindingType,
    string? Summary,
    CommitteeApprovalDetails? CommitteeApproval,
    List<string> Reviewers,
    IReadOnlyList<ChapterDetails> Chapters,
    IReadOnlyList<TranslationDetails> Translations,
    IReadOnlyList<FormatDetails> Formats
)
{
    public record AuthorDetails(string FirstName, string LastName);
    public record CommitteeApprovalDetails(bool IsApproved, string Feedback);
    public record ChapterDetails(string Title, string? Content);
    public record TranslationDetails(string Language, string Translator);
    public record FormatDetails(string FormatType, int TotalCopies, int SoldCopies);
}
